### ISSUE TITLE
Hero animations for AnimatedSwitcher and PageView

### ISSUE THREAD

**Issue by spkersten** [flutter/flutter#32972]
It would be nice to have Hero animations for `AnimatedSwitcher` and `PageView`. 

For flows where screen transitions are triggered by out side events or where transitions are reacting to a redux state, an `AnimatedSwitcher` seems to work better than a `Navigator`. Another use case of `AnimatedSwitcher` with heros is a tab view with a transition animation when switching and some common UI element between tabs (like a PersistentHeader, which might be collapsed on one tab but expanded on the next). The use cases for `PageView` are similar.

**Comment by BrutalCoding** [comment]
Would like to see this too

**Comment by spkersten** [comment]
I've created this package for `Hero` animations for `PageView`: https://pub.dev/packages/coast

**Comment by goderbauer** [comment]
cc @chunhtai Can this be de-duplicated into another issue?

**Comment by chunhtai** [comment]
yes this looks like a specific use case for https://github.com/flutter/flutter/issues/54200

closing as duplicate
